src/librustdoc/html/static/storage.js

```diff
@@ -27,20 +27,38 @@ function onEach(arr, func) {
 }
 
 function updateLocalStorage(name, value) {
-    if (typeof(Storage) !== "undefined") {
+    if (usableLocalStorage()) {
         localStorage[name] = value;
     } else {
         // No Web Storage support so we do nothing
     }
 }
 
 function getCurrentValue(name) {
-    if (typeof(Storage) !== "undefined" && localStorage[name] !== undefined) {
+    if (usableLocalStorage() && localStorage[name] !== undefined) {
         return localStorage[name];
     }
     return null;
 }
 
+function usableLocalStorage() {
```

Comment:
Can move this function above the two others please? It's just me being careful but better be safe than sorry. :)

Reply:
Try this code then:

```
foo();

function foo() {
    console.log('hello');
}
```